### ISSUE TITLE
Please do not allow installation with portalocker==2.0.0 (for maximum Python2.7 compatibility)

### ISSUE THREAD

**Issue by mpennington-te** [Preston-Landers/concurrent-log-handler#28]
[Python 2.7 is now end-of-life](https://www.python.org/doc/sunset-python-2/), but some of us support old linux distros which do not have a workable Python3.   In some instances, we cannot easily upgrade these systems.

Please consider changing your `setup.py` `install_requires` to `install_requires = ['portalocker<=1.7.1,>=1.4.0']`

Doing so prevents breaking those environments that are still stuck with Python2.7.

portalocker==2.0.0 fails in Python 2.7
-----------------------------------------

```python
(py27_u18) mpennington@Mpennington-X1:~$ pip freeze | grep -E "portalocker|concurrent"
concurrent-log-handler==0.9.16
portalocker==2.0.0
(py27_u18) mpennington@Mpennington-X1:~$ python -V
Python 2.7.17
(py27_u18) mpennington@Mpennington-X1:~$ python
Python 2.7.17 (default, Jul 20 2020, 15:37:01)
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import concurrent_log_handler
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mpennington/venv/py27_u18/local/lib/python2.7/site-packages/concurrent_log_handler/__init__.py", line 69, in <module>
    from portalocker import LOCK_EX, lock, unlock
  File "/home/mpennington/venv/py27_u18/local/lib/python2.7/site-packages/portalocker/__init__.py", line 5, in <module>
    from . import utils
  File "/home/mpennington/venv/py27_u18/local/lib/python2.7/site-packages/portalocker/utils.py", line 290
    def __init__(self, maximum: int, name: str = 'bounded_semaphore',
                              ^
SyntaxError: invalid syntax
>>> quit()
(py27_u18) mpennington@Mpennington-X1:~$
```

portalocker==1.7.1 works with Python 2.7
----------------------------------------------

```python
(py27_u18) mpennington@Mpennington-X1:~$ pip install portalocker==1.7.1
Collecting portalocker==1.7.1
  Using cached portalocker-1.7.1-py2.py3-none-any.whl (10 kB)
Installing collected packages: portalocker
  Attempting uninstall: portalocker
    Found existing installation: portalocker 2.0.0
    Uninstalling portalocker-2.0.0:
      Successfully uninstalled portalocker-2.0.0
Successfully installed portalocker-1.7.1
(py27_u18) mpennington@Mpennington-X1:~$ python
Python 2.7.17 (default, Jul 20 2020, 15:37:01)
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import concurrent_log_handler
>>> quit()
```

**Comment by Preston-Landers** [comment]
OK, that seems reasonable. I'm pretty sure we can set different requires based on Python 2 vs 3. I'll look at it soon. Thanks for the report.

**Comment by Preston-Landers** [comment]
The commit referenced above should resolve this. I plan to release a new version of Concurrent-Log-Handler within about a week, possibly with a few other small changes.

**Comment by Preston-Landers** [comment]
Should be fixed in release 0.9.17. Please let me know if that doesn't work. Thanks.
